org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/util/SonarLintWebView.java

```diff
@@ -72,60 +70,68 @@ public class SonarLintWebView extends Composite implements Listener, IPropertyCh
   private String htmlBody = "";
   private final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
   private ScheduledFuture<?> scheduledFuture;
+  private final Label labelToCopyColorsFrom;
 
   public SonarLintWebView(Composite parent, boolean useEditorFontSize) {
     super(parent, SWT.NONE);
-    this.useEditorFontSize = useEditorFontSize;
-    var layout = new GridLayout(1, false);
-    layout.marginWidth = 0;
+    var layout = new GridLayout();
+    layout.horizontalSpacing = 0;
     layout.marginHeight = 0;
-    this.setLayout(layout);
+    layout.marginWidth = 0;
+    layout.verticalSpacing = 0;
+    setLayout(layout);
+    this.labelToCopyColorsFrom = new Label(this, SWT.LEFT);
+    labelToCopyColorsFrom.setVisible(false);
+    var layoutData = new GridData();
+    layoutData.exclude = true;
+    labelToCopyColorsFrom.setLayoutData(layoutData);
+    this.useEditorFontSize = useEditorFontSize;
     try {
-      browser = new Browser(this, SWT.FILL);
+      browser = new Browser(this, SWT.NONE);
       addLinkListener(browser);
       browser.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 1, 1));
       browser.setJavascriptEnabled(false);
       // Cancel opening of new windows
       browser.addOpenWindowListener(event -> event.required = true);
       // Replace browser's built-in context menu with none
       browser.setMenu(new Menu(parent.getShell(), SWT.NONE));
-      updateColorAndFontCache();
-      this.defaultFont = getDefaultFont();
 
-      parent.getDisplay().addListener(SWT.Settings, this);
+      cacheColorsAndFonts();
+
+      parent.addListener(SWT.Paint, this);
       JFaceResources.getColorRegistry().addListener(this);
       JFaceResources.getFontRegistry().addListener(this);
 
-      this.addDisposeListener(e -> {
+      addDisposeListener(e -> {
         scheduler.shutdownNow();
         JFaceResources.getColorRegistry().removeListener(this);
         JFaceResources.getFontRegistry().removeListener(this);
-        parent.getDisplay().removeListener(SWT.Settings, this);
+        parent.removeListener(SWT.Paint, this);
       });
 
       // Hide the browser while resizing to avoid the lag
-      this.addControlListener(new ControlAdapter() {
+      addControlListener(new ControlAdapter() {
 
         @Override
         public void controlResized(ControlEvent e) {
-          browser.setVisible(false);
+          SonarLintWebView.this.setVisible(false);
```

Comment:
I don't understand that, how does `SonarLintWebView.this` work? Shouldn't just `this` be used?